### PR TITLE
Translate message.md to english

### PR DESCRIPTION
**Pull request by duvaneljulien**
But... it's not very _well written_.

### CODE REVIEW

File: en/message.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+### Enum
+
+Data **SHOULD** be represented as enumerations rather than as cryptic codes. In addition, enumeration positions **SHOULD** be serialized as a `camelCase` character to avoid mapping errors.
+
+```javascript
+Content-type: application/x.va.validation+json
+{
+    // No ambiguity
+    "title":"baron"
```
Comment (jfreyre) [inline en/message.md]: `,` missing (j'ai changé en fr)

File: en/message.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+```javascript
+Content-type: application/x.va.validation+json
+{
+    // By default, it is data
+    "myDateTime": "1997-09-02T19:20:30.45+01:00", 
+    // Is it long enough ? Explain when it is a displayable property
+    "myDateTimeDisplay": "Monday 2 September at 7pm 20mn 30sec",
+ 
+    "myDate": "1985-08-09", // By default, it is data
```
Comment (jfreyre) [inline en/message.md]: aussi sur deux ligne ?

File: en/basic-principles.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# Basic principles
```
Comment (yrandin) [inline en/basic-principles.md]: Core principles ?

File: en/basic-principles.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+## API First
+
+### Design API before implementation
+
+The signature of the API -- also called interface or contract -- **MUST** be done before the implementation (OpenAPI specification, Stub, etc).
+
+The goal is to allow people to give feedbacks as soon as possible and an internal discipline that focuses on :
+
+* the knowledge of the functional domain,
```
Comment (yrandin) [inline en/basic-principles.md]: Does the trailing comma make sense ? Is it MD markup ?

File: en/basic-principles.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+* entities and business resources, i.e., avoid having APIs for specific use-cases,
+* a real separation between the WHAT and the HOW
+
+**The API contract is the unique source of truth and not the implementation.** If you development langage does not support natively the generation of the documentation, then it is possible to write the documentation by hand.
```
Comment (yrandin) [inline en/basic-principles.md]: Typo: If you**r** development

File: en/basic-principles.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+An API **MUST** have a documentation in our enterprise Wiki that is a page with :
+
+* a description
+* the team in charge of the API
+* a link to the swagger documentation
+
+This wiki page **MUST** be added to our directory.
+
+_Nota Bene : Our API community of practice is aware that this solution is temporary. Eventually we are going to have a centralised directory such as an API Management._
```
Comment (yrandin) [inline en/basic-principles.md]: API Management tool ?

File: en/basic-principles.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+### Documentation
```
Comment (yrandin) [inline en/basic-principles.md]: Swagger documentation

File: en/basic-principles.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+An API **MUST** expose an explicite documentation, full and up to date of its endpoints and **SHOULD** expose it as a Swagger.
```
Comment (yrandin) [inline en/basic-principles.md]: typo: explicit

File: en/basic-principles.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+## REST
+
+### Ressources au lieu de verbe
+
+APIs **MUST** be designed around resources, and **MUST** not represent actions. An API **MAY** include hypermedia (HATEOAS).
+
+#### Maturity level
+
+Idealy, we aim for the second maturity level of Richardson, but it is possible to use the level 3. More information are available on https://martinfowler.com/articles/richardsonMaturityModel.html.
```
Comment (yrandin) [inline en/basic-principles.md]: Typo: Ideally

File: en/basic-principles.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+_Nota Bene : Our API community of practice is aware that this solution is temporary. Eventually we are going to have a centralised directory such as an API Management._
+
+### Documentation
+
+An API **MUST** expose an explicite documentation, full and up to date of its endpoints and **SHOULD** expose it as a Swagger.
+
+## REST
+
+### Ressources au lieu de verbe
```
Comment (yrandin) [inline en/basic-principles.md]: Ressources instead of verb ?

File: en/basic-principles.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+| Method   | Action    | Definition |          |
+|-----------|-----------|------------|----------|
+| POST      | Non-idempotent | Create a resource| C |
+| GET      | Nullipotent (Safe) | Get one (or many) resource(s) | R |
+| PUT      | Idempotent | Update a ressource | U |
+| DELETE      | Idempotent | Delete a resource | D |
+
+#### POST
+* A POST (create, in our case) executed with success returns a 201. The header has to contain `Location` with a link to the new entity.
+* In the case of an asynchronous operation, the response has to be 202 with an header `Location` in order to monitor the operation.
```
Comment (yrandin) [inline en/basic-principles.md]: Align with other sentences => with a 'Location' header

File: en/message.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+## Data and description
+
+### Encoding
+
+Data **SHOULD** be encoded in UTF-8.
+
+### Enum
+
+Data **SHOULD** be represented as enumerations rather than as cryptic codes. In addition, enumeration positions **SHOULD** be serialized as a `camelCase` character to avoid mapping errors.
```
Comment (yrandin) [inline en/message.md]: using 'camelCase' convention ?

File: en/message.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+### Common representation of business data
+
+The API **SHOULD** be based on common representation of business data. For more information, have a look at our Représentation communes des données business (Internal link).
+
+## Business validations
+
+### Format of business validations
+
+If the requests fails because of business validations, the response **SHOULD** use a 422 HTTP code, **SHOULD** have the following `Content-Type`
```
Comment (yrandin) [inline en/message.md]: have a clearly defined 'Content-Type' ?

File: en/message.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+ 
+            // Concerned field(s)
+            "fields":["firstName"],
+ 
+            // Variable data that are in the message (Validation property)
+            "valParams":{}
+        },
+        {
+            // Field translated according the i18n/l10n and displayable to the user
+            "display": "Le npa devrait comporter au moins 42 caractères",
```
Comment (yrandin) [inline en/message.md]: Why not translating the message ?

File: en/basic-principles.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+
+Standard HTTP methods have a meaning, they have to be used to determine the type of action to do.
+
+Although these methods are not equivalent to CRUD, it is preferable in our case to use them as such for simplification purposes and to keep only non idempotent creations.
+
+| Method   | Action    | Definition |          |
+|-----------|-----------|------------|----------|
+| POST      | Non-idempotent | Create a resource| C |
+| GET      | Nullipotent (Safe) | Get one (or many) resource(s) | R |
+| PUT      | Idempotent | Update a ressource | U |
```
Comment (ylerjen) [inline en/basic-principles.md]: Or create with a given id ?

**Pull request merged**